onnx/defs/schema.cc

```diff
@@ -829,63 +831,52 @@ const FunctionProto* OpSchema::GetFunction() const {
   return GetFunctionWithOpsetVersion(since_version_);
 }
 
-const FunctionProto*
-OpSchema::GetFunctionWithOpsetVersion(int opset_version, const std::string& domain, bool fail_on_invalid_op) const {
-  return const_cast<OpSchema*>(this)->GetFunctionWithOpsetInternal(opset_version, domain, fail_on_invalid_op);
+const FunctionProto* OpSchema::GetFunctionWithOpsetVersion(int requested_opset_version) const {
+  return const_cast<OpSchema*>(this)->GetFunctionWithOpsetInternal(requested_opset_version);
 }
 
-void OpSchema::ValidateReferencedOpsInFunciton(
+// when requesting a function at loading time,
+// requested_opset_version does not have to be the same as function_since_version.
+// When they are not the same, it is necessary to verify that ops used to define the function
+// are not updated between function_since_version and requested_opset_version (include requested_opset_version).
+// this call only validate ops in the default domain.
+// TODO: validate ops in other domains.
+bool OpSchema::ValidateReferencedOpsInFuncton(
     const FunctionProto* function,
     int requested_opset_version,
     int function_since_version,
-    const std::string& domain,
-    bool fail_on_invalid_op) {
+    std::set<std::string>* updated_ops) const {
+  bool has_no_invalid_op = true;
   if (requested_opset_version == function_since_version) {
-    // we validate that OpSchemas optained with requested_opset_version and function_since_version
-    // are the same for ops used in a function definition.
-    return;
+    return has_no_invalid_op;
   }
-  std::stringstream err;
-  bool has_invalid_op = false;
-  for (int node_index = 0; node_index < function->node_size(); ++node_index) {
-    const NodeProto& node = function->node(node_index);
-    const OpSchema* op1 = OpSchemaRegistry::Instance()->GetSchema(node.op_type(), requested_opset_version, domain);
-    const OpSchema* op2 = OpSchemaRegistry::Instance()->GetSchema(node.op_type(), function_since_version, domain);
+  for (auto& node : function->node()) {
+    const OpSchema* op1 =
```

Comment:
I think we should do the check below only if `node.domain()` is the standard domain? Or, should we assert that?

Reply:
However this validation works for ops in any domain, as long as the op is in the registry.

Reply:
But the parameter requested_opset_version specified is only for the standard onnx domain. It should not be used with a different domain

Reply:
True, let me add the condition.